src/main/frontend/src/components/loginAndRegistration/AuthContext.js

```diff
@@ -0,0 +1,16 @@
+import { createContext, useState } from 'react';
+
+export const AuthContext = createContext(null);
+
+export const AuthProvider = ({children}) => {
+  const[token, setToken] = useState(null);
+  let loggedIn = false;
+  if (token) {
+    loggedIn = true;
+  }
```

Comment:
this can be just `const loggedIn = !!token`. `!!` is short for checking if token exists and has non null value.